lib/angular/fieldLocation/fieldLocationController.js

```diff
@@ -12,22 +12,56 @@ function FieldLocationController($scope) {
   //TODO Repeating Fields
   fieldLocationCtrl.field.value = fieldLocationCtrl.field.values[0];
 
+  /**
+  * Attempt to set the current position in the input fields if the current position can be obtained.
+  * Otherwise set a default position and return a relevant error message.
+  */
   fieldLocationCtrl.setLocation = function(event) {
     event.stopPropagation();
 
-    //TODO Move it to a service to get the location
-    navigator.geolocation.getCurrentPosition(function(pos) {
+    /*
+    * Get the current position from the Geolocation Service.
+    */
+    GeolocationService.getGeolocation(function(err, response) {
+      if (response) {
+        $scope.$apply(function() {
+          // set the longitude/latitude
+          fieldLocationCtrl.field.value.lat = response.coords.latitude;
+          fieldLocationCtrl.field.value.long = response.coords.longitude;
+        });
+      } else {
+        $scope.$apply(function() {
+          // return a relevant error message in the UI
+          fieldLocationCtrl.field.err = err.message ? err.message : err;
+          // location not available, set a default
+          fieldLocationCtrl.field.value.lat = -1;
+          fieldLocationCtrl.field.value.long = -1;
```

Comment:
(-1, -1) is a valid latlong, we need to think of a better value if not `undefined` for an error state